File: src/main/java/io/dustin/finstage/financial/domain/IncomeStatement.java
package io.dustin.finstage.financial.domain;

import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.ToString;

import java.time.LocalDate;

@Getter
@Builder
@EqualsAndHashCode
@ToString
public class IncomeStatement {

    private final String symbol;
    private final LocalDate date;
    private final long revenue;
    private final long costOfRevenue;
    private final long grossProfit;
    private final long operatingIncome;
    private final long netIncome;
    private final double eps;

    private IncomeStatement(
            String symbol,
            LocalDate date,
            long revenue,
            long costOfRevenue,
            long grossProfit,
            long operatingIncome,
            long netIncome,
            double eps
    ) {
        this.symbol = symbol;
        this.date = date;
        this.revenue = revenue;
        this.costOfRevenue = costOfRevenue;
        this.grossProfit = grossProfit;
        this.operatingIncome = operatingIncome;
        this.netIncome = netIncome;
        this.eps = eps;
    }

    public static IncomeStatement of(
            String symbol,
            LocalDate date,
            long revenue,
            long costOfRevenue,
            long grossProfit,
            long operatingIncome,
            long netIncome,
            double eps
    ) {
        return new IncomeStatement(symbol, date, revenue, costOfRevenue, grossProfit, operatingIncome, netIncome, eps);
    }
}

File: src/main/kotlin/io/dustin/finstage/financial/domain/IncomeStatement.kt
package io.dustin.finstage.financial.domain

import java.time.LocalDate

data class IncomeStatement(
        val symbol: String,
        val date: LocalDate,
        val revenue: Long,
        val costOfRevenue: Long,
        val grossProfit: Long,
        val operatingIncome: Long,
        val netIncome: Long,
        val eps: Double
) {
    companion object {
        fun of(
                symbol: String,
                date: LocalDate,
                revenue: Long,
                costOfRevenue: Long,
                grossProfit: Long,
                operatingIncome: Long,
                netIncome: Long,
                eps: Double
        ): IncomeStatement {
            return IncomeStatement(
                    symbol,
                    date,
                    revenue,
                    costOfRevenue,
                    grossProfit,
                    operatingIncome,
                    netIncome,
                    eps
            )
        }
    }
}
